### PR TITLE
168 legal nav api updates

### PR DESCRIPTION
**Pull request by mitch-mitchel**
Draft PR for comparing code

### CODE REVIEW

File: ckeditor/plugins/a2j-guid/dialogs/guid.js [hunk 1]
```diff
@@ -200,9 +201,20 @@ CKEDITOR.dialog.add('guid', function(editor) {
           html: '<ul class="list-group"></ul>',
 
           commit: function (widget) {
-            if (selectedVariable) {
-              widget.setData('guid', selectedVariable.getAttribute('guid'))
-              widget.setData('name', selectedVariable.getHtml())
+            var dialog = CKEDITOR.dialog.getCurrent()
+            var selectedElement = dialog.getSelectedElement()
```
Comment (mitch-mitchel) [inline ckeditor/plugins/a2j-guid/dialogs/guid.js:206]: this is how the dialog grabs the source element that was double clicked

File: ckeditor/plugins/a2j-guid/dialogs/guid.js [hunk 2]
```diff
@@ -200,9 +201,20 @@ CKEDITOR.dialog.add('guid', function(editor) {
...
+            var url = selectedResource.getAttribute('guid')
+            var name = selectedResource.getHtml()
+
+            if (selectedResource) {
+              widget.setData('guid', url)
+              widget.setData('name', name)
+              // update the original source element's html on commit
+              if (selectedElement) {
+                // child 0 is the legal-nav-resource custom element
+                var sourceElement = selectedElement.getChild(0)
```
Comment (mitch-mitchel) [inline ckeditor/plugins/a2j-guid/dialogs/guid.js:216]: the 0th element is the legal-nav-resource-id and the 1 element is the span/wrapper that gives the little drag handle thing.

File: ckeditor/plugins/a2j-guid/plugin.js [hunk 3]
```diff
@@ -37,8 +36,8 @@
         },
 
         downcast: function(element) {
-          // Downcast set the html to be the guid
-          element.setHtml(this.data.guid)
+          // Downcast set the html to be the name of the resource
+          element.setHtml(this.data.name)
```
Comment (AnubhavDhingra) [inline ckeditor/plugins/a2j-guid/plugin.js:40]: This will reproduce the same error pointed out in #176

Comment (ghost) [inline ckeditor/plugins/a2j-guid/plugin.js:40]: Could reproduce the same error pointed out in #176

**Review by AnubhavDhingra — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
closes #168

**Pull request merged**